keras_hub/src/models/image_to_image.py

```diff
@@ -0,0 +1,351 @@
+        else:
+            inputs, input_is_scalar = normalize(inputs)
+
+        return inputs, input_is_scalar
+
+    def _normalize_generate_inputs(self, inputs):
+        """Normalize user input to the generate function.
+
+        This function converts all inputs to tensors, adds a batch dimension if
+        necessary, and returns a iterable "dataset like" object (either an
+        actual `tf.data.Dataset` or a list with a single batch element).
+        """
+        if tf and isinstance(inputs, tf.data.Dataset):
+            return inputs.as_numpy_iterator(), False
+
+        def normalize(x):
+            if isinstance(x, str):
+                return [x], True
+            if tf and isinstance(x, tf.Tensor) and x.shape.rank == 0:
+                return x[tf.newaxis], True
+            return x, False
+
+        if isinstance(inputs, dict):
+            for key in inputs:
+                inputs[key], input_is_scalar = normalize(inputs[key])
+        else:
+            inputs, input_is_scalar = normalize(inputs)
+
+        return inputs, input_is_scalar
+
+    def _normalize_generate_outputs(self, outputs, input_is_scalar):
+        """Normalize user output from the generate function.
+
+        This function converts all output to numpy with a value range of
+        `[0, 255]`. If a batch dimension was added to the input, it is removed
+        from the output.
+        """
+
+        def normalize(x):
+            outputs = ops.clip(ops.divide(ops.add(x, 1.0), 2.0), 0.0, 1.0)
+            outputs = ops.cast(ops.round(ops.multiply(outputs, 255.0)), "uint8")
+            outputs = ops.convert_to_numpy(outputs)
+            if input_is_scalar:
+                outputs = outputs[0]
+            return outputs
+
+        if isinstance(outputs[0], dict):
+            normalized = {}
+            for key in outputs[0]:
+                normalized[key] = normalize([x[key] for x in outputs])
+            return normalized
+        return normalize([x for x in outputs])
+
+    def generate(
+        self,
+        images,
+        inputs,
+        negative_inputs,
+        num_steps,
+        guidance_scale,
+        strength,
+        seed=None,
+    ):
+        """Generate image based on the provided `images` and `inputs`.
+
+        The `images` are reference images within a value range of `[-1.0, 1.0]`,
+        which will be resized to `self.backbone.height` and
+        `self.backbone.width`, then encoded into latent space by the VAE
+        encoder. The `inputs` are strings that will be tokenized and encoded by
+        the text encoder.
+
+        If `images` and `inputs` are a `tf.data.Dataset`, outputs will be
+        generated "batch-by-batch" and concatenated. Otherwise, all inputs will
+        be processed as batches.
+
+        Args:
+            images: python data, tensor data, or a `tf.data.Dataset`.
+            inputs: python data, tensor data, or a `tf.data.Dataset`.
```

Comment:
please add more details about this arg `inputs`, seems too generic.

Reply:
Thanks for the review. I have some ideas to refine the args for `TextToImage`, `ImageToImage` and `Inpaint`.
I believe it would be better to align the API design with that of `CausalLM`.

Working on it!

Reply:
@divyashreepathihalli 
I have refactored the arguments for all `TextToImage`, `ImageToImage` and `Inpaint` tasks to use a single `inputs` as the inputs, which aligns with `CausalLM`.

For `inputs`:
- TextToImage: Accepts `str`, `[str]` , `{"prompts": str|[str]}`, `{"prompts": str|[str], "negative_prompts": str|[str]}`, and `tf.data.Dataset`.
- ImageToImage: Accepts `{"images": array, "prompts": str|[str]}`, `{"images": array, "prompts": str|[str], "negative_prompts": str|[str]}`, and`tf.data.Dataset`.
- Inpaint: Accepts `{"images": array, "masks": array, "prompts": str|[str]}`, `{"images": array, "masks": array, "prompts": str|[str], "negative_prompts": str|[str]}` and `tf.data.Dataset`.

I’ve also noticed that `Flux` doesn't support `"negative_prompts"`. To address this, I’ve introduced a new attr, `support_negative_prompts`, that toggles this feature based the attr.

Please let me know if this looks good.